### PR TITLE
[Merged by Bors] - Fix `doc_markdown` lints in `bevy_ecs`

### PR DESCRIPTION
**Pull request by mfdorst**
#3457 adds the `doc_markdown` clippy lint, which checks doc comments to make sure code identifiers are escaped with backticks. This causes a lot of lint errors, so this is one of a number of PR's that will fix those lint errors one crate at a time.

This PR fixes lints in the `bevy_ecs` crate.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_ecs/src/query/iter.rs [hunk 1]
```diff
@@ -35,7 +35,7 @@ where
     /// This does not check for mutable query correctness. To be safe, make sure mutable queries
     /// have unique access to the components they query.
     /// This does not validate that `world.id()` matches `query_state.world_id`. Calling this on a `world`
-    /// with a mismatched WorldId is unsound.
+    /// with a mismatched [`WorldId`] is unsound.
```
Comment (killercup) [inline crates/bevy_ecs/src/query/iter.rs:38]:
```suggestion
    /// with a mismatched [`WorldId`][crate::world::WorldId] is unsound.
```

(This is a private item)

File: crates/bevy_ecs/src/query/iter.rs [hunk 2]
```diff
@@ -174,8 +174,8 @@ where
     /// # Safety
     /// This does not check for mutable query correctness. To be safe, make sure mutable queries
     /// have unique access to the components they query.
-    /// This does not validate that `world.id()` matches `query_state.world_id`. Calling this on a `world`
-    /// with a mismatched WorldId is unsound.
+    /// This does not validate that `world.id()` matches `query_state.world_id`. Calling this on a
+    /// `world` with a mismatched [`WorldId`] is unsound.
```
Comment (killercup) [inline crates/bevy_ecs/src/query/iter.rs:178]:
```suggestion
    /// `world` with a mismatched [`WorldId`][crate::world::WorldId] is unsound.
```

(This is also a private item)

**Review by killercup — APPROVED** [review]
LGTM

Not sure if we care: `cargo doc --open --document-private-item` complains about some broken links in private items. I commented on two issues introduced here, another on is on `World::flush` (change `[Archetype]` to ```[`Archetype`][crate::archetype::Archetype]```, and ad tick marks around `[Component]`).

**Comment by alice-i-cecile** [comment]
Thanks for the heads up; we can handle that separately.

**Pull request closed without merging**